### PR TITLE
calendar_screen first commit

### PR DESCRIPTION
**Pull request by dalia2112**

### CODE REVIEW

File: app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    @BindView(R.id.calendar_month_button)
+    Button mCalendarMonthButton;
+    @BindView(R.id.calendar_list)
+    RecyclerView mCalendarRecycleView;
+    @BindView(R.id.switch_my_events)
+    Switch mSwitchEvents;
+    @BindView(R.id.filters)
+    Button mFiltersButton;
+    @BindView(R.id.projects_list_parent)
```
Comment (gabrielblaj07) [inline app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java]: wrong naming

File: app/src/main/res/layout/fragment_calendar.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        android:layout_marginStart="@dimen/calendar_day_list_margin"
+        android:layout_marginTop="@dimen/calendar_day_list_margin"
+        android:layout_marginEnd="@dimen/calendar_day_list_margin"
+        android:orientation="horizontal"
+        android:scrollbars="none"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/calendar_month_button" />
+
+    <Switch
```
Comment (gabrielblaj07) [inline app/src/main/res/layout/fragment_calendar.xml]: Make the switch look like in the design

File: app/src/main/res/layout/fragment_calendar.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        android:layoutDirection="rtl"
+        android:stateListAnimator="@null"
+        android:text="@string/filter_button_text"
+        android:textColor="@color/white"
+        android:textSize="@dimen/calendar_text_size"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/calendar_list" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/projects_list_parent"
```
Comment (gabrielblaj07) [inline app/src/main/res/layout/fragment_calendar.xml]: wrong naming

File: app/src/main/res/layout/item_event.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        android:layout_marginTop="@dimen/event_item_details_margin"
+        android:drawableStart="@drawable/ic_event_tags"
+        android:drawablePadding="@dimen/event_item_drawable_padding"
+        android:fontFamily="@font/roboto_regular"
+        android:textColor="@color/secondary"
+        android:textSize="@dimen/event_item_details_text_size"
+        app:layout_constraintStart_toEndOf="@id/event_time"
+        app:layout_constraintTop_toBottomOf="@id/event_title" />
+
+    <ProgressBar
```
Comment (gabrielblaj07) [inline app/src/main/res/layout/item_event.xml:71]: the progress bar should have round corners

File: app/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -21,11 +21,12 @@
...
     <color name="pale_violet">#d0c9d6</color>
     <color name="primary">#de411b</color>
-    <color name="primary_transparent">#ccde411b</color>
+    <color name="primary_transparent">#0dde411b</color>
     <color name="gray5">#e0e0e0</color>
     <color name="white">#ffffff</color>
     <color name="gray4">#bdbdbd</color>
-    <color name="gray3">#828282</color>
+    <color name="gray4_transparent">#CCbdbdbd</color>
+    <color name="grey3">#828282</color>
```
Comment (gabrielblaj07) [inline app/src/main/res/values/colors.xml]: 😞

File: app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java [hunk 6]
```diff
@@ -38,9 +42,19 @@ public CalendarDayAdapter(Context context, List<CalendarDay> tags) {
     @Override
     public CalendarViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
         View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_calendar_day, parent, false);
+        view.getLayoutParams().width =  getRecyclerWidth() / 5;
         return new CalendarViewHolder(view);
     }
 
+    private int getRecyclerWidth() {
```
Comment (gabrielblaj07) [inline app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java:47]: Nice one 😮

File: app/src/main/res/drawable/progress_bar_shape.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@android:id/background">
+        <shape>
+            <corners android:radius="8dp" />
```
Comment (gabrielblaj07) [inline app/src/main/res/drawable/progress_bar_shape.xml]: Once upon a time there was a file called dimens.xml

File: app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+public class CalendarDayAdapter extends RecyclerView.Adapter<CalendarDayAdapter.CalendarViewHolder> {
+
+    public static final String TRAINING_EVENT = "Training";
+    public static final String CONFERENCE_EVENT = "Conference";
+    public static final String CERTIFICATION_EVENT = "Certification";
+
+    private final Context mContext;
+
+    private List<CalendarDay> mCalendarDayList;
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java]: as discussed with Gabi also, I would like to renounce the mStuff naming and continue without the m prefix. there is an extra line between context and list

File: app/src/main/res/drawable/switch_thumb_selector.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_checked="false">
+        <shape android:dither="true" android:shape="rectangle" android:useLevel="false" android:visible="true">
+            <solid android:color="#ffffff" />
+            <corners android:radius="100dp" />
+            <size android:width="18dp" android:height="18dp" />
+            <stroke android:width="4dp" android:color="#0000ffff" />
```
Comment (gabrielblaj07) [inline app/src/main/res/drawable/switch_thumb_selector.xml]: Once upon a time there was a file called dimens.xml (and colors.xml as well)

File: app/src/main/res/drawable/switch_track_selector.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_checked="false">
```
Comment (gabrielblaj07) [inline app/src/main/res/drawable/switch_track_selector.xml:3]: Once upon a time there was a file called dimens.xml

File: app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java (hunk 8 above)
Comment (grauralex92) [reply app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java]: @soundrob be aware of the naming discussion here

File: app/src/main/java/com/endava/myendava/adapters/EventAdapter.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+import com.endava.myendava.R;
+import com.endava.myendava.models.Event;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class EventAdapter extends RecyclerView.Adapter<EventAdapter.EventViewHolder> {
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/EventAdapter.java]: Event**s**Adapter? 😛

File: app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java (hunk 6 above)
Comment (dalia2112) [reply app/src/main/java/com/endava/myendava/adapters/CalendarDayAdapter.java:47]: ..........................................

File: app/src/main/java/com/endava/myendava/adapters/EventAdapter.java [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+    public static final String TRAINING_EVENT = "Training";
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/EventAdapter.java]: duplicate code ... it exist also in the other adapter. please extract in from the adapters in a common place

File: app/src/main/java/com/endava/myendava/adapters/EventAdapter.java [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    public static final String CONFERENCE_EVENT = "Conference";
+    public static final String CERTIFICATION_EVENT = "Certification";
+
+    private Context mContext;
+
+    private List<Event> mEvents;
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/EventAdapter.java]: same as in the other adapter

File: app/src/main/java/com/endava/myendava/adapters/EventAdapter.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+            mEventTitle.setText(event.getTitle());
+            mEventTime.setText(event.getDuration());
+            mEventTags.setText(event.getFormattedTags());
+            mEventNrPeople.setText(event.getParticipantsNr() + " of 10");
+            int f = (int) ((event.getParticipantsNr() / 10.0) * 100);
+            mProgressBar.setProgress(f);
+            mEventType.setBackgroundTintList(mContext.getColorStateList(getEventColor(event.getType())));
+        }
+
+        private int getEventColor(String event) {
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/EventAdapter.java]: duplicate code

File: app/src/main/res/drawable/progress_bar_shape.xml (hunk 7 above)
Comment (dalia2112) [reply app/src/main/res/drawable/progress_bar_shape.xml]: 😭

File: app/src/main/java/com/endava/myendava/adapters/EventParentAdapter.java [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.endava.myendava.R;
+import com.endava.myendava.models.EventParent;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class EventParentAdapter extends RecyclerView.Adapter<EventParentAdapter.ProjectParentViewHolder> {
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/adapters/EventParentAdapter.java]: no need for this adapter ... please see https://medium.com/@saber.solooki/sticky-header-for-recyclerview-c0eb551c3f68

File: app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+    SwitchCompat mSwitchEvents;
+    @BindView(R.id.filters)
+    Button mFiltersButton;
+    @BindView(R.id.events_list_parent)
+    RecyclerView mProjectsRecyclerView;
+    @BindView(R.id.progress_bar)
+    ProgressBar mProgressBar;
+
+    private EventParentAdapter mParentEventAdapter;
+
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java:58]: don't use the mNaming anymore ... also there is an extra blank space 😛

File: app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+        Event event2 = new Event("Certification title", "1h 30", tags, 2, "Certification");
+        Event event3 = new Event("Conference title", "2h", tags, 9, "Conference");
+        Event event4 = new Event("Training title", "3h 30", tags, 5, "Training");
+        List<Event> list1 = new ArrayList<>();
+        List<Event> list2 = new ArrayList<>(Arrays.asList(event1, event2, event3));
+        List<Event> list3 = new ArrayList<>(Arrays.asList(event1, event2, event3, event4));
+        return new ArrayList<>(Arrays.asList(list1, list2, list3));
+    }
+
+    private List<CalendarDay> getCalendarDayList() {
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/fragments/CalendarFragment.java]: move data generation to another file which can be easily deleted after real implementation

File: app/src/main/java/com/endava/myendava/models/CalendarDay.java [hunk 18]
```diff
@@ -0,0 +1,43 @@
+package com.endava.myendava.models;
+
+import java.util.List;
+
+public class CalendarDay {
+    private int dayNr;
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/models/CalendarDay.java]: space needed

Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/models/CalendarDay.java]: don't use Nr or other shortcuts ... it makes the code more easily to read if you are using full words

File: app/src/main/java/com/endava/myendava/models/Event.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
+package com.endava.myendava.models;
+
+import java.util.List;
+
+public class Event {
+
+    private String title;
+    private String duration;
+    private List<Tag> tags;
+    private int participantsNr;
```
Comment (grauralex92) [inline app/src/main/java/com/endava/myendava/models/Event.java]: same as for dayNr

**Review by grauralex92 — CHANGES_REQUESTED** [review]

**Review by grauralex92 — APPROVED** [review]

**Pull request merged**